lib/ReactViews/Map/Panels/SharePanel/SharePanel.tsx

```diff
@@ -50,522 +32,89 @@ interface PropTypes extends WithTranslation {
-              }}
-            >
-              {t("share.printViewButton")}
-            </button>
-          </div>
-        </div>
-        <hr className={Styles.thinLineDivider} />
-        <div
-          className={classNames(DropdownStyles.section, Styles.advancedOption)}
-        >
-          <div className={Styles.btnWrapper}>
-            <button
-              type="button"
-              onClick={this.toggleAdvancedOptions}
-              className={Styles.btnAdvanced}
-            >
-              <span>{t("share.btnAdvanced")}</span>
-              {this.advancedIsOpen() ? (
-                <Icon glyph={Icon.GLYPHS.opened} />
-              ) : (
-                <Icon glyph={Icon.GLYPHS.closed} />
-              )}
-            </button>
-          </div>
-          {this.advancedIsOpen() && (
-            <>
-              <Box className={DropdownStyles.section}>
-                {this.isUrlShortenable() && (
-                  <div className={Styles.advancedOption}>
-                    <Checkbox
-                      textProps={{ small: true }}
-                      id="shortenUrl"
-                      isChecked={(this.shouldShorten() as boolean) ?? false}
-                      onChange={this.onShortenClicked}
-                      className={Styles.checkbox}
-                    >
-                      <TextSpan>{t("share.shortenUsingService")}</TextSpan>
-                    </Checkbox>
-                  </div>
-                )}
-                <div className={Styles.advancedOption}>
-                  <Checkbox
-                    textProps={{ small: true }}
-                    id="includeStory"
-                    title="Include Story in Share"
-                    isChecked={this.state.includeStoryInShare}
-                    onChange={this.onIncludeStoryInShareClicked}
-                    className={Styles.checkbox}
-                  >
-                    <TextSpan>{t("includeStory.message")}</TextSpan>
-                  </Checkbox>
-                </div>
-              </Box>
-              <hr className={Styles.thinLineDivider} />
-              <div className={DropdownStyles.section}>
-                <TextSpan className={Styles.paragraph}>
-                  {t("share.embedTitle")}
-                </TextSpan>
-                <Input
-                  large
-                  dark
-                  type="text"
-                  readOnly
-                  placeholder={this.state.placeholder}
-                  value={iframeCode}
-                  onClick={e => {
-                    const target = e.target as HTMLInputElement;
-                    return target.select();
-                  }}
-                />
-              </div>
-            </>
-          )}
-        </div>
-      </div>
-    );
-  }
-
-  openWithUserClick() {
-    if (this.props.onUserClick) {
-      this.props.onUserClick();
+        </Box>
+      );
+    } else {
+      return (
+        <SharePanelContent
```

Comment:
While this reduces the lines of code in the SharePanel class, it is a bit confusing. If we think about SharePanel having three modes `catalogShare`, `storyShare`, and `printEmbedShare`, it is confusing for someone looking at this class that `primtEmbedShare` (the default one) is created in such a different way to the other two. My thinking was to just have `SharePanel` and `ShareUrl`, and you could move a lot of functionality out of `SharePanel` into `ShareUrl`. But having so much logic in `SharePanelContent` is confusing structurally. Can you refactor so that the code in `SharePanelContent` is contained within `SharePanel` and `ShareUrl`?

Reply:
All good for @steve9164 so all good for me!